espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/GsonParserHelpers.kt

```diff
@@ -0,0 +1,22 @@
+package io.appium.espressoserver.lib.helpers
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+
+class GsonParserHelpers {
+
+    inline fun <reified T : Enum<T>> parseEnum(jsonObj: JsonObject, propName: String, helperMessage: String = ""): T? {
+        val property = jsonObj.get(propName)
+        if (property != null) {
+            val propValueAsString = jsonObj.get(propName).asString.toUpperCase()
+            try {
+                return enumValueOf<T>(propValueAsString)
+            } catch (e: Exception) {
```

Comment:
can the exception class be narrowed down?

Reply:
Yep. Changed it to catch `IllegalArgumentException`